### PR TITLE
Add membership() to User object

### PR DESCRIPTION
**Pull request by enmand**
Add paged `GroupMembership`s list to the Box User object

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @enmand, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

**Comment by enmand** [comment]
CLA signed

**Comment by boxcla** [comment]
Verified that @enmand has just signed the CLA. Thanks, and we look forward to your contribution.

File: boxsdk/object/user.py [hunk 1]
```diff
@@ -1,11 +1,45 @@
...
 from .base_object import BaseObject
+from .group_membership import GroupMembership
 
 
 class User(BaseObject):
     """Represents a Box user."""
 
     _item_type = 'user'
+
+    def groups(self, starting_index=0, limit=100, include_page_info=False):
```
Comment (jmoldow) [inline boxsdk/object/user.py]: This is called `groups`, but it returns `GroupMembership`s, not `Group`s.

Idea: rename this method to `membership` (then it mirrors `Group.membership()`). Optionally, add another method

```python
def groups(self, starting_index=0, limit=100, include_page_info=False):
    for group_membership, current_page_size, index_in_current_page in self.membership(starting_index=starting_index, limit=limit, include_page_info=True):
        group = group_membership.group
        if include_page_info:
            yield group, current_page_size, index_in_current_page
        else:
            yield group
```

File: boxsdk/object/user.py [hunk 2]
```diff
@@ -1,11 +1,45 @@
...
+        """
+        A generator over all of the Groups that this User is a member of.
```
Comment (jmoldow) [inline boxsdk/object/user.py]: Lets make it clearer that these are `GroupMembership`s, not `Group`s.

**Comment by jmoldow** [comment]
Looks good to me, except for the comments I'd like to see addressed. But @Jeff-Meadows  and @potrebic , can you also take a look at this?

Also, are you able to add tests? I'd like to see tests before we merge this.

**Comment by enmand** [comment]
Sure, I'll add some tests.

Since `User.membership()` and `Group.membership()` are essentially the same, would it be better to have them inherit a common object which implements most of the `membership` func?

**Comment by Jeff-Meadows** [comment]
Thanks for the PR.

* +1 for tests
* Can you add info about this change to HISTORY.rst?
* Optionally, add yourself to AUTHORS.rst

**Comment by mattwiller** [comment]
We added [`user.get_group_memberships()`](https://box-python-sdk.readthedocs.io/en/latest/boxsdk.object.html#boxsdk.object.user.User.get_group_memberships) to the SDK, which supersedes this change

**Pull request closed without merging**